plugins/lsp/language-servers/rust-analyzer-config.nix

```diff
@@ -1,167 +1,65 @@
-      enum ? null,
-      minimum ? null,
-      maximum ? null,
-      items ? null,
-      anyOf ? null,
-      uniqueItems ? null,
-      # unused, but for exhaustivness
-      enumDescriptions ? null,
-    }@def:
-    if enum != null then
-      types.enum enum
-    else if anyOf != null then
-      types.oneOf (builtins.map (mkRustAnalyzerType opt) anyOf)
-    else if builtins.isList type then
-      if builtins.length type == 2 && builtins.head type == "null" then
-        types.nullOr (mkRustAnalyzerType opt (def // { type = builtins.elemAt type 1; }))
-      else
-        types.oneOf (builtins.map (t: mkRustAnalyzerType opt (def // { type = t; })) type)
-    else if type == "boolean" then
-      types.bool
-    else if type == "object" then
-      types.attrsOf types.anything
-    else if type == "string" then
-      types.str
-    else if type == "null" then
-      nullType
-    else if type == "array" then
-      types.listOf (
-        mkRustAnalyzerType "${opt}-items" (
-          if items == null then throw "null items with array type (in ${opt})" else items
-        )
-      )
-    else if type == "number" then
+    { kind, ... }@typeInfo:
+    if kind == "enum" then
+      lib.types.enum typeInfo.values
+    else if kind == "oneOf" then
+      lib.types.oneOf (lib.map mkRustAnalyzerType typeInfo.subTypes)
+    else if kind == "list" then
+      lib.types.listOf (mkRustAnalyzerType typeInfo.item)
+    else if kind == "number" then
+      let
+        inherit (typeInfo) minimum maximum;
+      in
       if minimum != null && maximum != null then
-        types.numbers.between minimum maximum
+        lib.types.numbers.between minimum maximum
       else if minimum != null then
-        types.addCheck types.number (x: x >= minimum)
+        lib.types.addCheck lib.types.number (x: x >= minimum)
       else if maximum != null then
-        types.addCheck types.number (x: x <= maximum)
+        lib.types.addCheck lib.types.number (x: x <= maximum)
       else
-        types.number
-    else if type == "integer" then
+        lib.types.number
+    else if kind == "integer" then
+      let
+        inherit (typeInfo) minimum maximum;
+      in
       if minimum != null && maximum != null then
-        types.ints.between minimum maximum
+        lib.types.ints.between minimum maximum
       else if minimum != null then
-        types.addCheck types.int (x: x >= minimum)
+        lib.types.addCheck lib.types.int (x: x >= minimum)
       else if maximum != null then
-        types.addCheck types.int (x: x <= maximum)
+        lib.types.addCheck lib.types.int (x: x <= maximum)
       else
-        types.int
+        lib.types.int
+    else if kind == "object" then
+      lib.types.attrsOf lib.types.anything
+    else if kind == "string" then
+      lib.types.str
+    else if kind == "boolean" then
+      lib.types.bool
     else
-      throw "unhandled type `${builtins.toJSON type}` (in ${opt})";
+      throw "Unknown type: ${kind}";
```

Comment:
Doesn't look like any of these have `rawLua` support?

Reply:
(can be a TODO if this isn't a regression)

Reply:
Yeah I forgot to add that TODO